### PR TITLE
For #25401 - Pocket sponsored stories telemetry

### PR DESCRIPTION
**Pull request by Mugurell**
Work based on a yet unmerged patch which adds a setting used for the telemetry functionality also.
Will rebase when possible.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Mugurell** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
  * This data will help us understand how the new Pocket sponsored stories feature is used.
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * This data will help us understand which features in the home screen are more important for users. 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No. 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
An indication of when the Pocket sponsored stories feature is enabled | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/25401
An indication of when certain sponsored story is shown to the user | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/25401
An indication for how many times the total number that story was shown | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/25401
An indication for the position of the shown story in list | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/25401
An indication of when the user clicks a certain sponsored story | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/25401
An indication for how many times the clicked story was shown | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/25401
An indication for the position of the clicked story in list | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/25401

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix
7. How long will this data be collected?
  * The new probes are set to expire in version 105 and 106 along following the same expiration date as other related probes. Renewal may be considered at that point.
8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean. 
11. Where do you intend to share the results of your analysis?
  *  Results will be shared through Glean with the mobile team and with the sponsored stories provider. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

**Review by mcarare — APPROVED** [review]
r+ code review.

**Comment by Mugurell** [comment]
Rebased following recent merges.

File: app/metrics.yaml [hunk 1]
```diff
@@ -5375,6 +5394,59 @@ pocket:
...
+          How many times was this story shown, including current.
+      position:
+        type: string
+        description: |
+          Position of the clicked story in the list shown.
+          Uses the [row x column] matrix notation.
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/25401
+    data_reviews:
+      - ?????
```
Comment (rocketsroger) [inline app/metrics.yaml:5423]: Make sure you update this (applies to the other ones as well) before merging

**Pull request merged**